provider/resource_playbook.go

```diff
@@ -0,0 +1,574 @@
+	}
+
+	playFirstTime, okay := data.Get("play_first_time").(bool)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'play_first_time'!", ansiblePlaybook)
+	}
+
+	if playFirstTime || replayable {
+		args := []string{}
+
+		// Get the rest of args
+		for _, arg := range argsTf {
+			tmpArg, okay := arg.(string)
+			if !okay {
+				log.Fatal("ERROR [ansible-playbook]: couldn't assert type: string")
+			}
+
+			args = append(args, tmpArg)
+		}
+
+		runAnsiblePlay := exec.Command(ansiblePlaybookBinary, args...)
+
+		runAnsiblePlayOut, runAnsiblePlayErr := runAnsiblePlay.CombinedOutput()
+
+		if runAnsiblePlayErr != nil {
+			playbookFailMsg := fmt.Sprintf("ERROR [ansible-playbook]: couldn't run ansible-playbook\n%s! "+
+				"There may be an error within your playbook.\n%v",
+				playbook,
+				runAnsiblePlayErr,
+			)
+			if !ignorePlaybookFailure {
+				log.Fatal(playbookFailMsg)
+			} else {
+				log.Print(playbookFailMsg)
+			}
+		}
+
+		log.Printf("LOG [ansible-playbook]: %s", runAnsiblePlayOut)
+
+		if err := data.Set("play_first_time", false); err != nil {
+			log.Fatal("ERROR [ansible-playbook]: couldn't set 'play_first_time'!")
+		}
+
+		// Wait for playbook execution to finish, then remove the temporary file
+		err := runAnsiblePlay.Wait()
+		if err != nil {
+			log.Printf("LOG [ansible-playbook]: didn't wait for playbook to execute: %v", err)
+		}
+
+		providerutils.RemoveFile(tempInventoryFile)
+	}
+
+	return nil
+}
+
+func resourcePlaybookUpdate(data *schema.ResourceData, meta interface{}) error {
+	originalID := data.Id()
+
+	data.SetId(originalID + "-taint")
+
+	name, okay := data.Get("name").(string)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'name'!", ansiblePlaybook)
+	}
+
+	groups, okay := data.Get("groups").([]interface{})
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'groups'!", ansiblePlaybook)
+	}
+
+	argsTf, okay := data.Get("args").([]interface{})
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'args'!", ansiblePlaybook)
+	}
+
+	args := []string{}
+
+	for _, arg := range argsTf {
+		tmpArg, okay := arg.(string)
+		if !okay {
+			log.Fatal("ERROR [ansible-playbook]: couldn't assert type: string")
+		}
+
+		args = append(args, tmpArg)
+	}
+
+	inventoryFileNamePrefix := ".inventory-"
+
+	createdTempInventory := providerutils.BuildPlaybookInventory(inventoryFileNamePrefix+"*.ini", name, -1, groups)
+	if err := data.Set("temp_inventory_file", createdTempInventory); err != nil {
+		log.Fatal("ERROR [ansible-playbook]: couldn't set 'temp_inventory_file'!")
+	}
+
+	// Get all available temp inventories and pass them as args
+	inventories := providerutils.GetAllInventories(inventoryFileNamePrefix)
+
+	log.Print("[INVENTORIES]:")
+	log.Print(inventories)
+
+	for _, inventory := range inventories {
+		args = append(args, "-i", inventory)
+	}
+
+	if err := data.Set("args", args); err != nil {
+		log.Fatalf("ERROR [ansible-playbook]: couldn't set 'args'! %s", err)
+	}
```

Comment:
This is adding a non-existent inventory file for every subsequent run. You can see this if you run an apply several times on the same project and then look at the `ansible_playbook` resource in the terraform state.

Reply:
Yes, I think that must be because after, ``terraform apply`` finishes, those temporary inventory files are removed from the system temp directory (line 506), but arguments are not refreshed - hence, you'll see temporary inventory files in terraform state in ``ansible_playbook`` resource. They existed during the apply, but not after.